src/sets.jl

```diff
@@ -13,12 +13,18 @@ for (T, f) in [
     end
 end
 
-Polymake.pm_Set(s::AbstractSet{S}) where S = pm_Set(collect(s))
-
 pm_Set{T}(v::Vector) where T = pm_Set(Vector{T}(v))
 pm_Set{T}(s::Set) where T = pm_Set{T}(collect(s))
 
-pm_Set(::Type{T}) where T = pm_Set{T}()
+pm_Set{S}(n::T) where {S,T <: Integer} = Polymake.scalar2set(S(n))
+
+pm_Set{T}(itr) where T = union!(pm_Set{T}(), itr)
+
+function pm_Set(itr)
+    # use IteratorEltype(itr) trait?
+    T = typeof(first(itr))
```

Comment:
This can lead to problems if the iterator is stateful. We could loose the first element then.

Reply:
I think it would be fine to rely on `eltype(itr)`

Reply:
yeah, I had this with `eltype` before, but
```julia
julia> eltype(1:5)
Int64

julia> eltype(x for x in 1:5)
Any
```
julia Sets do
```julia
Set(itr) = _Set(itr, IteratorEltype(itr))

_Set(itr, ::HasEltype) = Set{eltype(itr)}(itr)

function _Set(itr, ::EltypeUnknown)
    T = @default_eltype(itr)
    (isconcretetype(T) || T === Union{}) || return grow_to!(Set{T}(), itr)
    return Set{T}(itr)
end
```
is `T = @default_eltype(itr)` good enough?

Reply:
No idea what `T = @default_eltype(itr)` is doing but probably yes.

Reply:
`@edit Base.@default_eltype(x for x in 1:5)`

Reply:
well, not good enough, `@default_eltype` may return unions, etc:
```
Construction: Error During Test at /home/kalmar/.julia/dev/PolymakeWrap/test/sets.jl:271
  Got exception outside of a @test
  MethodError: no method matching PolymakeWrap.Polymake.pm_Set{Union{Int32, Int64}}()
  Closest candidates are:
    PolymakeWrap.Polymake.pm_Set{Union{Int32, Int64}}(!Matched::Array{T,1} where T) where T at /home/kalmar/.julia/dev/PolymakeWrap/src/sets.jl:16
...
```

Reply:
For new also can go the `Set(collect(iter))` way instead. Not optimal, but whatever.

Reply:
At leas if `eltype(T)` is not a concrete type.